designs/placement-groups.md

```diff
@@ -0,0 +1,65 @@
+# Placement Group Configuration
+
+## Goals
+* Allowing a user to configure EC2 Placement Group strategies from an `EC2NodeClass` resource
+* Allow the Placement Group strategies to be developed independently from one another
+
+## Background
+At present, there is no way to configure an [EC2 Placement Group](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/placement-groups.html) with Karpenter. The addition of this feature will expose an important feature of EC2 to Karpenter users, expanding the set of use-cases the tool is appropriate for.
+
+## Proposed Design
+To enable the configuration of EC2 Placement Groups with Karpenter, I propose the addition of a `.spec.placementStrategy` value in the `karpenter.k8s.aws/v1beta1/EC2NodeClass` resource. This new value will have three child fields --  `cluster`, `partition`, and `spread` -- only one of which can be enabled for a given `EC2NodeClass` instance, which is in line with the EC2 API. 
+
+Given the simple and self-contained nature of EC2 Placement Group configurations, my sense is that it makes litle sense to enable the import of preexisting Placement Group configurations. This would add implementation complexity without actually enabling additional features for the user. Instead, Karpenter should completely manage the lifecycle of EC2 Placement Group resources, tying their lifetimes to the lifetime of the `.spec.placementStrategy` value in the supervising EC2NodeClass resource.
```

Comment:
Can you expand on this a bit? I don't think it's immediately clear to the reader why that's the case. I can see the argument that placement groups (at least for spread and partition) assume you are defining spread at the instance-level rather than having the flexibility of containers which allows you to spread your application out on a level just below the instance-level.